fix: show error on Editor composer textarea

wa0x6e opened the pull request:
### Summary

<!-- Related issues, a description or list of the changes and the motivation behind them -->

Laying the groundworks for #517 

The current proposal creation form does not validate the body, it just set a hard limit on the body textarea. 

In addition, the composer editor textarea is not styled exactly like the rest of the textarea, preventing using the same css to style the errors.

This PR will:

- add error support for the composer textarea, and validating the proposal body and user space statement
- refactor the editor textarea, to make use of the existing textarea styling

### Test 

- Go to the proposal creation page
- Fill the textarea over the 9600 chars limit
- It should show the error 

![Screenshot 2024-07-23 at 23 38 02](https://github.com/user-attachments/assets/cd2eef5a-bcd5-4054-b917-9b0f926a35e3)

The PREVIEW markdown tab default height has also been adjusted to match the textarea height by default

![Screenshot 2024-07-23 at 23 38 43](https://github.com/user-attachments/assets/cb1fcbb3-6b52-4a1d-b7fb-7fcf39754d61)

Sekhmet reviewed (commented):
If I have textarea with error and switch to markdown preview and back error state disappears.

wa0x6e commented:
> If I have textarea with error and switch to markdown preview and back error state disappears.

Fixed.

Also added validation to the user space statement

Sekhmet commented on apps/ui/src/components/Ui/Composer.vue:
```diff
@@ -74,14 +115,34 @@ const editor = useMarkdownEditor(
+      ref="editorRef"
+      v-model.trim="model"
+      class="s-input h-[200px]"
+      @input="event => (model = (event.target as HTMLInputElement).value)"
```
As it's now `v-model` we don't need `@input` anymore I think.

bonustrack commented:
Can we show a different limit on space using Turbo? It should be 40k for Turbo, also for normal spaces I believe the limit should be 10k

wa0x6e commented:
> Can we show a different limit on space using Turbo? It should be 40k for Turbo, also for normal spaces I believe the limit should be 10k

Yes, but this will come from a different PR #524

bonustrack reviewed (approved):
tACK

Sekhmet reviewed (approved):
tACK
